analyzers/src/SonarAnalyzer.CSharp/Rules/SupplyParameterFromQueryNeedRoutableComponent.cs

```diff
@@ -0,0 +1,64 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules.CSharp;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class SupplyParameterFromQueryNeedRoutableComponent : SonarDiagnosticAnalyzer
```

Comment:
@cristian-ambrosini-sonarsource Since I merged the rule with S6797, I renamed the class to `BlazorQueryParameterRoutableComponent`.
Open to other suggestion

Reply:
Fine for me! I knew that the class would be renamed anyway. I'll let you do that in your PR (I'll not address your comment Antonio if it's fine for you because it will be renamed anyway shortly after)